test/e2e/scripts/run_tests.sh

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
```

Comment:
Suggest adding a user check (is root) to `run_tests.sh`
Support running as non-root user as well aka gate within the python module based on check result in run_tests.sh

I got it to run on fedora36, but all tests failed, because of `Operation not permitted` errors, I suspect it's because of the missing sudo?

Once it works on my fedora36 box can also test it on my other ubuntu box.

Reply:
Right, I knew I was missing something, I'll take a look and add some gates.

Reply:
I ended up adding a check directly as a fixture in the python tests, it will fail if the effective uid is not 0 and prevent anything else from running.

Reply:
I can say that it works fine :rofl: